### DISCUSSION TITLE
Notify sender on message being held

### DISCUSSION THREAD

**Discussion by klebed** [postalserver/postal#1526] [Feature suggestions]
Hi!

I'm using postal as smtp gateway for sending external messages initially originated from mail server. It's solving some issues and doing it's job very well, but when sender mistyped or recipient in suppression list sender won't know it, because postal would silently held his message.

I suggest adding an option to notify sender that message has been held with the reason, so sender could take some actions, for example check if address is correct, or read that user not exist any more in target system, or there was a rate limit... 
As additional suggestion, these notifications could contain unique links so user could push message for second attempt (useful with rate limit) or just resend message with new recipient address.

**Comment by willpower232** [comment]
There is already a webhook for notifying when a message has been held https://docs.postalserver.io/developer/webhooks#message-status-events

I'm not sure how well it will work with SMTP as you won't necessarily know which message it was.

**Reply by klebed** [reply]
If you held a message, you know who is the sender, it's title and its reference. I believe there is all info needed to not only notify sender, but even make him able to react and decide what to do with the message.

Webhook is good, but it require to develop own application for that, while notification of sender seems essential to me and better have it out of the box.
